### PR TITLE
fix(test): improved test coverage and documentation

### PR DESCRIPTION
**Pull request by sethyates**
## What does this Pull Request do?

Takes the test coverage to nearly 100% and the documentation coverage to 100%.
## Why is this change being made?

Belts and braces.
## Where should the reviewer start?
1. esdoc.json
2. manual/Overview.md
3. manual/Installation.md
4. manual/Example.md
5. src/GPT.js
6. then through the remaining src/\* files
7. then through the remaining test/\* files
## How was this tested? How can the reviewer verify your testing?
1. running `npm run tdd:coverage` until the coverage reached as close to 100% as practical
2. running `npm run doc` until the document coverage reached 100%
## Completion checklist
- [X] The pull request has been appropriately labelled according to our conventions.
- [X] Documentation has been updated.
- [X] The change has unit & integration tests as appropriate.

### CODE REVIEW

File: src/GPT.js [hunk 1]
```diff
@@ -123,38 +147,40 @@ export default class GPT {
...
    * companion slots. Calling this API clears the ad and removes the slot object
-   * from the internal state maintained by GPT. Calling any more functions on
+   * from the internal state maintained by {@link GPT}. Calling any more functions on
    * that slot object will result in undefined behaviour. Note the browser may
    * still not free the memory associated with that slot if a reference to it is
    * maintained by the publisher page. Calling this API makes the div associated
    * with that slot available for reuse.
    *
-   * @param {Array<!Slot>=} optSlots The array of slots to
+   * @param {Array<!Slot>} [optSlots] The array of slots to
```
Comment (jnewman) [inline src/GPT.js]: 👍 I prefer this syntax

File: src/Service.js [hunk 2]
```diff
@@ -62,28 +72,35 @@ export default class Service {
...
-      const id = `${slot.getAdUnitPath()}_${index}`;
+      this._gpt._removeSlot(slot);
+      const id = `${slot.getAdUnitPath()}_${index + 1}`;
       delete this._slotIdMap[id];
     }
   }
 
   /**
-   * UNDOCUMENTED - Returns a map of ID's to slots.
+   * Returns a map of ID's to slots.
```
Comment (jnewman) [inline src/Service.js]: Nit - `IDs`

File: src/Service.js [hunk 3]
```diff
@@ -62,28 +72,35 @@ export default class Service {
...
    *
-   * @returns {Object<string, Slot>} The map of ID's to slots.
+   * @experimental
+   * @returns {object} The map of ID's to slots.
```
Comment (jnewman) [inline src/Service.js]: Same nit as above

Comment (jnewman) [inline src/Service.js]: Should `object` not be `Object`? I see it capped here: http://usejsdoc.org/tags-typedef.html

File: src/Size.js [hunk 4]
```diff
@@ -1,32 +1,90 @@
...
   getHeight() {
-    return this.h;
+    return this.height;
+  }
+
+  /**
+   * Returns whether the {@link Size} is empty.
+   *
+   * @experimental
+   * @type {boolean} True if the size is empty.
```
Comment (jnewman) [inline src/Size.js]: Nit - you were using `true` in previous PRs

File: src/Slot.js [hunk 5]
```diff
@@ -4,27 +4,29 @@ import ImpressionViewableEvent from './events/ImpressionViewableEvent';
...
+   * Creates a new {@link Slot}.
    *
    * @param {string} adUnitPath Full path of the ad unit with the network code and unit code.
-   * @param {GeneralSize=} size Width and height of the added slot. This is the
+   * @param {GeneralSize} [size] Width and height of the added slot. This is the
    * size that is used in the ad request if no responsive size mapping is provided
    * or the size of the viewport is smaller than the smallest size provided in the mapping.
-   * @param {string=} optDiv ID of the div that will contain this ad unit.
+   * @param {string} [optDiv] ID of the div that will contain this ad unit.
+   * @param {number} [instance = 0] The instance count of the slot.
```
Comment (jnewman) [inline src/Slot.js]: Too bad JsDoc can't figure out the default

File: test/unit/TargetingMap.spec.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
+import TargetingMap from '../../src/TargetingMap';
+
+/** @test {TargetingMap} */
+describe('TargetingMap', () => {
+  /** @test {TargetingMap#get} */
+  describe('#get', () => {
+    it('returns empty array if not set', () => {
+      const map = new TargetingMap();
```
Comment (jnewman) [inline test/unit/TargetingMap.spec.js]: Nit (minor) - could build a map in before each or use a higher level function to build the test.

**Review by jnewman — APPROVED** [review]
🚀 🌟

File: README.md [hunk 7]
```diff
@@ -8,7 +8,6 @@
 [![Dependencies](https://img.shields.io/david/dev/krux/gpt-mock.svg)](./package.json)
 [![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
-[![Gitter](https://badges.gitter.im/krux/gpt-mock.svg)](https://gitter.im/krux/gpt-mock)
```
Comment (sethyates) [inline README.md]: No. Not for this :-)

**Pull request merged**